### PR TITLE
Add FAQ

### PR DESCRIPTION
**Pull request by odedponcz**
not connected to README yet

### CODE REVIEW

File: doc/faq.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+
+## What is the license model of Dragonfly? Is it an open source?
+Dragonfly is released under [BSL 1.1](../LICENSE.md) (Business Source License). The BSL license is more permissive than AGPL-like licenses in some elements. In general terms, it means that dragonfly's code is free to use and free to change as long as you do not provide paying support or managed services for an in-memory datastore.
+We followed the trend of other technological companies like Elastic, Redis, MongoDB, Cockroach labs, Redpanda Data to protect our rights to provide service and support for the software we are building. 
+
+## Can I use dragonfly in production?
+License wise you are free to use dragonfly in your production as long as you do not provide dragonfly as a managed service. From a code maturity point of view, Dragonfly's code is tested with unit testing and also against the Redis testing TCL scripts. However as with any new software there are use cases that are hard to test and predict. We advise you to run your own particular use case on dragonfly for a few days before considering production usage.
```
Comment (romange) [inline doc/faq.md:15]: `as long as you do not sell services directly related to dragonfly or in-memory datastores` .

File: doc/faq.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+## I get it, but why not change Redis open source?
+Many of the pain points we addressed like Multi-core support, memory optimization, forkless snapshot, improved eviction algo, unit testing and more... have been the achilles heel and a long standing requests by the community. Some were not addressed for as long as ten years, others were dismissed because they require a complete rewrite of the engine. We believe that innovation at this scale is impossible on Redis OSS. KeyDB reused Redis OSS to the limit and introduce multi-threaded architecture into it. But to do all that with minimal locking we had to think about all things holistically. How you handle backpressure? how you do snapshotting? How you implement multi-key operations or blocking transactions? For that you need  need special algorithms to provide atomicity, you need fibers/coroutines to be able to block your calling context yet unblock the cpu for other tasks etc. All this was designed bottom up from scratch. BTW, Dragonfly does use some Redis code around single value data-structures like zset, hset etc...
```
Comment (romange) [inline doc/faq.md:21]: Can you pls just leave the title? I think the answer is too verbose . I will add the answer some time this week.

**Review by romange — CHANGES_REQUESTED** [review]

File: doc/faq.md (hunk 1 above)
Comment (odedponcz) [reply doc/faq.md:15]: Do we have covrage stats?

Comment (romange) [reply doc/faq.md:15]: No, and there is definitely a place for improvement there in terms of coverage. 
But the thing is that we can do high-precision testing at all levels.

File: doc/faq.md (hunk 2 above)
Comment (odedponcz) [reply doc/faq.md]: Its shorter now. If you still think it should not get in I will delete.

Comment (romange) [reply doc/faq.md]: I also do not wish to grill Redis in our repo. yeah, pls delete.

File: doc/faq.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+## I get it, but why not change Redis open source?
+It's impossible to reuse Redis OSS to make it multi-threaded besides what KeyDB did. With multi-threading you need to think about all things holistically. How you handle backpressure? how you do snapshotting? How you implement multi-key operations or blocking transactions? For that you need special algorithms to provide atomicity, you need fibers/coroutines to be able to block your calling context yet unblock the cpu for other tasks etc. All this was designed bottom up from scratch. BTW, Dragonfly does use some Redis code around single value data-structures like zset, hset etc...
+
+## If only Dragonfly had this command I would use it for sure
+Dragonfly implements ~130 Redis commands which we think represent a good coverage of the market. However this is not based empirical data. Having said that, if you have commands that are not covered, please feel free to open an issue for that or vote for an existing issue. We will do our best to prioritise those commands according to their popularity.
```
Comment (romange) [inline doc/faq.md:24]: you should delete "However this is not based empirical data.".

Comment (romange) [inline doc/faq.md:24]: vote for *the* existing issue.

**Review by romange — COMMENTED** [review]
it's mostly fine. I gave few comments below

File: doc/faq.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+## I get it, but why not change Redis open source?
+It's impossible to reuse Redis OSS to make it multi-threaded besides what KeyDB did. With multi-threading you need to think about all things holistically. How you handle backpressure? how you do snapshotting? How you implement multi-key operations or blocking transactions? For that you need special algorithms to provide atomicity, you need fibers/coroutines to be able to block your calling context yet unblock the cpu for other tasks etc. All this was designed bottom up from scratch. BTW, Dragonfly does use some Redis code around single value data-structures like zset, hset etc...
```
Comment (romange) [inline doc/faq.md:21]: `t's impossible to reuse Redis OSS to make it multi-threaded` - too opinionated. 
I will use material from here but I prefer to write this paragraph myself.

**Pull request merged**